doc/man3/SSL_CTX_set_session_ticket_cb.pod

```diff
@@ -0,0 +1,145 @@
+SSL_SESSION_get0_ticket_appdata,
+SSL_SESSION_set1_ticket_appdata - manage session ticket application data
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ typedef enum {
+     TICKET_FATAL_ERR_MALLOC,
+     TICKET_FATAL_ERR_OTHER,
+     TICKET_NONE,
+     TICKET_EMPTY,
+     TICKET_NO_DECRYPT,
+     TICKET_SUCCESS,
+     TICKET_SUCCESS_RENEW
+ } TICKET_RETURN;
+
+ void SSL_CTX_set_session_ticket_cb(SSL_CTX *ctx,
+                                    tls_generate_session_ticket_cb_fn gen_cb,
+                                    tls_decrypt_session_ticket_cb_fn dec_cb,
+                                    void *arg);
+
+ int SSL_SESSION_set1_ticket_appdata(SSL_SESSION *ss, const void *data, size_t len);
+ void SSL_SESSION_get0_ticket_appdata(SSL_SESSION *ss, void **data, size_t *len);
+
+=head1 DESCRIPTION
+
+SSL_CTX_set_set_session_ticket_cb() sets the application callbacks B<gen_cb>
+and B<dec_cb> that are a server to set and get application data stored with
+a session, and placed into a session ticket. The value of B<arg> is passed
+to the callbacks.
+
+B<gen_cb> is the application defined callback invoked when a session ticket is about
+to be created. The application may call SSL_SESSION_set1_ticket_appdata() at this
+time to add application data to the session ticket. The value of B<arg> is the same
+as that given to  SSL_CTX_set_session_ticket_cb(). The B<gen_cb> callback is defined
+as type B<tls_generate_session_ticket_cb_fn>.
+
+ typedef int (*tls_generate_session_ticket_cb_fn)(SSL *s, void *arg);
+
+B<dec_cb> is the application defined callback invoked after a session ticket has been
+decrypted and any session ticket application data is available. The application may
+call SSL_SESSION_get_ticket_appdata() at this time to retrieve the application data.
+The value of B<arg> is the same as that given to  SSL_CTX_set_session_ticket_cb().
+The B<dec_cb> callback is defined as type B<tls_decrypt_session_ticket_cb_fn>.
+
+ typedef TICKET_RETURN (*tls_decrypt_session_ticket_cb_fn)(SSL *s, SSL_SESSION *ss,
+                                                          const unsigned char *keyname,
+                                                          TICKET_RETURN retv,
+                                                          void *arg);
+
+SSL_SESSION_set1_ticket_appdata() sets the application data specified by B<data> and
+B<len> into B<ss> which is then placed into any generated session tickets. It may be
+called at any time before a session ticket is created to update the data placed into
+the session ticket. However, given that sessions and tickets are created by the
+handshake, the B<gen_cb> is provided to notify the application that a session ticket
+is about to be generated.
+
+SSL_SESSION_get0_ticket_appdata() assigns B<data> to the session ticket application
+data and assigns B<len> to the length of the session ticket application data from
+B<ss>, which may be set via SSL_SESSION_set1_ticket_appdata() or by application data
```

Comment:
Not a fan of having the object be the subject of the next sentence; basically repeating the word.